README.md

```diff
@@ -0,0 +1,215 @@
+* A2 has a built-in `apostrophe-images` "slideshow" widget type, while A3 only has a built-in single-image `@apostrophecms/image` widget type. By default `apostrophe-images` will be upgraded to `@apostrophecms/image`, with only the first image present in each. However you can use the `mapWidgetTypes` option, documented below, to override this mapping during the upgrade.
+* Since A3 does not have a built-in cropping feature yet, there is currently no accommodation for it in A3's `@apostrophecms/image` widget type. However an attempt is made to carry over the cropping data in the format which is expected to work in A3 in the near future.
+
+## Installation
+
+This tool installs as a module **inside your existing A2 project.** This is necessary to gain access to information such as the schemas of your existing piece and page and widget types.
+
+```bash
+cd my-existing-a2-project
+npm install @apostrophecms/content-upgrader
+```
+
+## Configuration
+
+Edit `app.js` of your A2 project. Add the module to the `modules` section:
+
+```javascript
+modules: {
+  // ... all of your existing A2 modules here ...
+  '@apostrophecms/content-upgrader': {}
+}
+```
+
+Then create `lib/modules/@apostrophecms/content-upgrader/index.js`. Here you can optionally address any content transformations and set the default locale, which is important *even if you have no immediate plans to localize your site in other languages.*
+
+### Setting the default locale
+
+In `lib/modules/@apostrophecms/content-upgrader/index.js`, be sure to set the default locale name to match your A3 project. Otherwise your site may appear to have no content after the upgrade.
+
+If no locale configuration at all is done in your A3 project, it will be `en`, so that is the default here as well if you do not specify otherwise.
+
+```javascript
+// In lib/modules/@apostrophecms/content-upgrader/index.js
+module.exports = {
+  defaultLocale: 'fr'
+};
+```
+
+### Transforming doc types
+
+In A2, the names of piece type modules themselves and the `name` option configured for the module were often different. By convention the piece type module name was plural and the `name` option was singular. The `name` option is what is used in the database to set the `type` property of each piece.
+
+Yes, this was confusing. That's why in A3, the `name` option no longer exists, and **the name of the module and the `type` property in the database are always the same.** Also, by convention, module names in A3 are singular.
+
+What does this mean for us when we upgrade the content? It means that most of the time, **in our A3 source code we need to change the module name to singular and remove its `name` option, but we don't need to change the `type` in the database,** because the new module's name will likely match it.
```

Comment:
I don't like the idea of trying to change anything (as in mutating the module name value to be singular). Maybe I'm misunderstanding, but we should simply use the configured `name` value for the new module name.